c_emulator/riscv_sim.c

```diff
@@ -1075,3 +1109,56 @@ int main(int argc, char **argv)
   flush_logs();
   close_logs();
 }
+
+int rv_cfg_c_ext_enable(char * isa_str, char * ext_pattern)
+{
+  pcre2_code          *re;
+  int                 errornumber;
+  PCRE2_SIZE          erroroffset;
+  pcre2_match_data    *match_data;
+  int                 rc;
+
+  re = pcre2_compile(
+    ext_pattern,            /* the pattern */
+    PCRE2_ZERO_TERMINATED,  /* indicates pattern is zero-terminated */
+    0,                      /* default options */
+    &errornumber,           /* for error number */
+    &erroroffset,           /* for error offset */
+    NULL);                  /* use default compile context */
+
+  if (re == NULL) {    /* Compilation failed */
+    PCRE2_UCHAR buffer[256];
+    pcre2_get_error_message(errornumber, buffer, sizeof(buffer));
+    fprintf(stderr, "PCRE2 compilation failed at offset %d: %s\n", (int)erroroffset,
+            buffer);
+  }    
+
+  match_data = pcre2_match_data_create_from_pattern(re, NULL);
+
+  rc = pcre2_match(
+    re,                   /* the compiled pattern */
+    isa_str,              /* the subject string */
+    strlen(isa_str),      /* the length of the subject */
+    0,                    /* start at offset 0 in the subject */
+    0,                    /* default options */
+    match_data,           /* block for storing the result */
+    NULL);                /* use default match context */
+
+  if ((rc == 0) || (rc == -1)) {  /* Does not match */
+    return(0);
+  }
+  else if (rc == 1) {
+    return(1);
```

Comment:
Small nit, why not make the return type bool ?

Reply:
Agreed.  Done.